### ISSUE TITLE
Local Pick up and Free shipping problem

### ISSUE THREAD

**Issue by beckyware** [woocommerce/woocommerce#17524]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [x] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. I have local pickup enabled for a certain shipping zone (south carolina) but it will not show up at check out
2. to try and bypass this problem, I went and made a free shipping coupon to give out to those who may need it. I followed all relevant instructions online and this option did not work either.
3.

## Expected/actual behavior

When I follow those steps, I see...

I was expecting to see...

## Isolating the problem

- [ ] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin here.

`
### WordPress Environment ###

Home URL: https://creativewarehouse.org
Site URL: https://creativewarehouse.org
WC Version: 3.2.2
Log Directory Writable: ✔
WP Version: 4.8.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 5.6.30
PHP Post Max Size: 65 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.45.0
OpenSSL/1.0.1e

SUHOSIN Installed: –
MySQL Version: 5.6.32
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.2
WC Database Prefix: wp_mmzd66rhjt_
Total Database Size: 3.03MB
Database Data Size: 2.64MB
Database Index Size: 0.39MB
wp_mmzd66rhjt_woocommerce_sessions: Data: 0.01MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_order_items: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_order_itemmeta: Data: 0.01MB + Index: 0.01MB
wp_mmzd66rhjt_woocommerce_tax_rates: Data: 0.00MB + Index: 0.01MB
wp_mmzd66rhjt_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.01MB
wp_mmzd66rhjt_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_commentmeta: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_comments: Data: 0.00MB + Index: 0.01MB
wp_mmzd66rhjt_links: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_ngg_album: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_ngg_gallery: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_ngg_pictures: Data: 0.00MB + Index: 0.00MB
wp_mmzd66rhjt_options: Data: 1.06MB + Index: 0.05MB
wp_mmzd66rhjt_postmeta: Data: 0.48MB + Index: 0.16MB
wp_mmzd66rhjt_posts: Data: 1.07MB + Index: 0.08MB
wp_mmzd66rhjt_termmeta: Data: 0.00MB + Index: 0.01MB
wp_mmzd66rhjt_terms: Data: 0.00MB + Index: 0.01MB
wp_mmzd66rhjt_term_relationships: Data: 0.00MB + Index: 0.01MB
wp_mmzd66rhjt_term_taxonomy: Data: 0.00MB + Index: 0.01MB
wp_mmzd66rhjt_usermeta: Data: 0.01MB + Index: 0.01MB
wp_mmzd66rhjt_users: Data: 0.00MB + Index: 0.01MB
wp_mmzd66rhjt_woocommerce_termmeta: Data: 0.00MB + Index: 0.00MB

### Post Type Counts ###

attachment: 107
display_type: 7
displayed_gallery: 1
feedback: 11
jetpack_migration: 2
jp_img_sitemap: 1
jp_sitemap: 1
jp_sitemap_master: 1
nav_menu_item: 6
nectar_slider: 6
ngg_gallery: 2
ngg_pictures: 4
page: 13
portfolio: 61
post: 3
product: 3
product_variation: 2
revision: 506
shop_coupon: 2
shop_order: 6
shop_order_refund: 1
wpcf7_contact_form: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (9) ###

Contact Form 7: by Takayuki Miyoshi – 4.9.1
Jetpack by WordPress.com: by Automattic – 5.4
Salient Visual Composer: by Michael M - WPBakery.com | Modified by ThemeNectar – 4.11.2
Go Daddy Quick Setup: by GoDaddy.com
LLC – 1.06

Simple Lightbox: by Archetyped – 2.6.0
WooCommerce PayPal Powered by Braintree Gateway: by WooCommerce – 2.0.4 – Not tested with the active version of WooCommerce
WooCommerce Stripe Gateway: by WooCommerce – 3.2.3 – Not tested with the active version of WooCommerce
WooCommerce Services: by Automattic – 1.8.2 – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.2.2

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #728 - /shop/
Cart: #729 - /cart/
Checkout: #730 - /checkout/
My account: #731 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Salient
Version: 7.0.8
Author URL: http://themenectar.com
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Archive Template: Your theme has a woocommerce.php file
you will not be able to override the woocommerce/archive-product.php custom template since woocommerce.php has priority over archive-product.php. This is intended to prevent display issues.

Overrides: salient/woocommerce/cart/cart-shipping.php version 2.5.0 is out of date. The core version is 3.2.0
salient/woocommerce/cart/cart-totals.php
salient/woocommerce/cart/cart.php version 2.3.8 is out of date. The core version is 3.1.0
salient/woocommerce/cart/shipping-calculator.php version 2.0.8 is out of date. The core version is 3.2.0
salient/woocommerce/checkout/form-checkout.php
salient/woocommerce/checkout/form-shipping.php version 2.2.0 is out of date. The core version is 3.0.9
salient/woocommerce/checkout/review-order.php
salient/woocommerce/content-product.php version 2.5.0 is out of date. The core version is 3.0.0
salient/woocommerce/content-single-product.php version 1.6.4 is out of date. The core version is 3.0.0
salient/woocommerce/loop/add-to-cart.php version 2.5.0 is out of date. The core version is 3.0.0
salient/woocommerce/single-product/product-image.php version 2.0.14 is out of date. The core version is 3.1.0
salient/woocommerce/single-product/tabs/description.php
salient/woocommerce/single-product/tabs/tabs.php

Outdated Templates: ❌Learn how to update
`
```
</details>

**Comment by mikejolley** [comment]
This can happen with incorrectly configured zones. e.g. if they overlap and another zone matches before your local pickup zone.

Remember you can **add more than one method to each zone**, and that **they match from top to bottom**. _If you have a US zone, make sure it is below your carolina zone._

Feel free to post screenshots so I can point out the err, or contact support. This isn't a bug as far as I can tell my side.

Thanks

**Comment by beckyware** [comment]
I did what you suggested... it worked for about 3 minutes... now it's still having the same problem

![screen shot 2017-11-02 at 1 06 57 pm](https://user-images.githubusercontent.com/33321946/32339713-fe6dd91a-bfce-11e7-904f-074377d429f3.png)
![screen shot 2017-11-02 at 1 07 39 pm](https://user-images.githubusercontent.com/33321946/32339720-0135329c-bfcf-11e7-8c6a-b3ccefaff720.png)

**Comment by mikejolley** [comment]
Is united states still selected under that top zone? Edit and show me.

**Comment by beckyware** [comment]
I don't think so... unless I'm missing something...
![screen shot 2017-11-02 at 2 25 18 pm](https://user-images.githubusercontent.com/33321946/32343414-b7a19566-bfd9-11e7-9589-7332842c090c.png)

**Comment by mikejolley** [comment]
US is not selected in zone regions.

**Comment by beckyware** [comment]
Sorry- I miss read that. Thought you said NOT selected.... Do you mean like this? if so... same thing is happening... 
![screen shot 2017-11-02 at 2 41 59 pm](https://user-images.githubusercontent.com/33321946/32344132-2bf97db4-bfdc-11e7-87fe-ef4c0fbd0295.png)


![screen shot 2017-11-02 at 2 42 25 pm](https://user-images.githubusercontent.com/33321946/32344133-2c08024e-bfdc-11e7-9a19-3503315f4fca.png)

**Comment by mikejolley** [comment]
And it's one zip per line. You have them comma separated.

**Comment by beckyware** [comment]
That was it! Thanks... one more question. I've created a coupon code that takes 100% off the item price and is supposed to activate free shipping... when I apply the coupon (beckygetsitforfree), the shipping price remains... then I added the free shipping zone, and have it at the very top of the list and it works beautifully. However, should I remove the coupon in my cart, all my shipping options diss
![screen shot 2017-11-02 at 3 19 51 pm](https://user-images.githubusercontent.com/33321946/32345705-62d5ade4-bfe1-11e7-9878-6c9316243619.png)
apear

![screen shot 2017-11-02 at 2 58 17 pm](https://user-images.githubusercontent.com/33321946/32345325-364b5392-bfe0-11e7-9d22-af4b7d29adcc.png)

![screen shot 2017-11-02 at 2 58 41 pm](https://user-images.githubusercontent.com/33321946/32345331-3d3eeaf6-bfe0-11e7-88c7-acc88ccfaaf7.png)

![screen shot 2017-11-02 at 3 09 06 pm](https://user-images.githubusercontent.com/33321946/32345383-5dc4566c-bfe0-11e7-8be3-39c62cdd8a13.png)
